ios/Classes/SwiftDevCycleFlutterClientSdkPlugin.swift

```diff
@@ -4,278 +4,355 @@ import UIKit
+                    .build(onInitialized: { error in
+                        if (error != nil) {
+                            callbackArgs["error"] = "\(String(describing: error))"
+                        }
+                        self.channel.invokeMethod("clientInitialized", arguments: callbackArgs)
+                    })
+            }
+            result(nil)
+        case "identifyUser":
+            if let dvcUser = user {
+                try? self.dvcClient?.identifyUser(user: dvcUser, callback: { error, variables in
+                    callbackArgs["callbackId"] = callbackId
+                    if (error != nil) {
+                        callbackArgs["error"] = "\(String(describing: error))"
+                    } else {
+                        callbackArgs["variables"] = self.variablesToMap(variables: variables ?? [:])
+                    }
+                    self.channel.invokeMethod("userIdentified", arguments: callbackArgs)
+                })
+            }
+            result(nil)
+        case "resetUser":
+            try? self.dvcClient?.resetUser(callback: { error, variables in
+                callbackArgs["callbackId"] = callbackId
+                if (error != nil) {
+                    callbackArgs["error"] = "\(String(describing: error))"
+                } else {
+                    callbackArgs["variables"] = self.variablesToMap(variables: variables ?? [:])
+                }
+                self.channel.invokeMethod("userReset", arguments: callbackArgs)
+            })
+            result(nil)
+        case "allFeatures":
+            result(featuresToMap(features: self.dvcClient?.allFeatures() ?? [:]))
+        case "allVariables":
+            result(variablesToMap(variables: self.dvcClient?.allVariables() ?? [:]))
+        case "getPlatformVersion":
+            result("iOS " + UIDevice.current.systemVersion)
+        case "variable":
+            return getResultForVariable(args: args, callbackArgs: callbackArgs, result: result)
+        case "track":
+            if let event = args?["event"] as? [String: Any], let dvcEvent = getEventFromDict(dict: event) {
+                self.dvcClient?.track(dvcEvent)
+            }
+        case "flushEvents":
+            self.dvcClient?.flushEvents(callback: { error in
+                callbackArgs["callbackId"] = callbackId
+                if (error != nil) {
+                    callbackArgs["error"] = "\(String(describing: error))"
+                }
+                self.channel.invokeMethod("eventsFlushed", arguments: callbackArgs)
+            })
+            result(nil)
+        default:
+            result(FlutterMethodNotImplemented)
+        }
     }
     
-    if let privateCustomData = dict["privateCustomData"] as? [String: Any] {
-      userBuilder.privateCustomData(privateCustomData)
+    private func getResultForVariable(args: [String: Any]?, callbackArgs: [String:Any], result: @escaping FlutterResult) {
```

Comment:
new method to figure out what type of variable it is to call the proper DVCVariable